lib/poxa/registry.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Poxa.Registry do
+
```

Comment:
I would probably just define the private function `adapter` and directly use

`adapter.register(channel)`